django-stubs/db/models/fields/related.pyi

```diff
@@ -204,10 +205,9 @@ class OneToOneField(ForeignKey[_ST, _GT]):
     @overload
     def __get__(self, instance: Any, owner: Any) -> Self: ...
 
-class ManyToManyField(RelatedField[_ST, _GT]):
-    _pyi_private_set_type: Sequence[Any]
-    _pyi_private_get_type: RelatedManager[Any]
+_To = TypeVar("_To", bound=Model)
 
+class ManyToManyField(RelatedField[Any, Any], Generic[_To, _M]):
```

Comment:
Can we make it generic on all the things and simplify our plugin a bit?

Reply:
Which parts are you referring to?

The plugin only acts on what is `_T` and `_M` here, essentially the `to=` and `through=` argument on `ManyToManyField`.

It does however have to sort some stuff out when there was no explicit `through=` argument passed, as Django automatically generates a through model behind the scenes in that case.

Additionally it supports resolving lazy references e.g. `"myapp.MyModel"` and replaces either `_T` or `_M` with any found model type.

Reply:
`_GT, _ST`? Will it help?

Reply:
No, this changes so that those arguments aren't used at all any more, for `ManyToManyField`. It's just the parent class that wants them.

The generics of the descriptor class does almost all the work now instead (except when `through` is implicit, then the plugin does that part)

Reply:
It's a bit hard to follow, but instead of using `_GT` and `_ST` and let our plugin do _all_ the work, I've updated the descriptor class `ManyToManyDescriptor` to become generic over a model, and that will be the `through=` model for a `ManyToManyField()`.

And if you now look a bit further, you find: `ManyToManyDescriptor.through -> type[_M]` which means that the `through=` argument align with the `MyModel.m2m_field.through` without any plugin work (again, except for special cases of lazy referencing or not passing the `through=` argument, which is all the new plugin code you see in this PR)

Reply:
Corresponding part is the `to=` argument, plugin code will _only_ be running for lazy referencing. Previously this was completely managed by the plugin, now it's instead mypy doing the work.